### DISCUSSION TITLE
Beautify json and xml requests payloads.

### DISCUSSION THREAD

**Discussion by tomasz80** [AnWeber/httpyac#800] [General]
Hello,
I have started using httpyac as rest/soap client - it is fantastic, but I have a question regarding formatting (beautifying) request's payloads: JSON and XML.
When I am using vs code command "Format Document" on http file, vs code is complaining: "There is no formatter for 'http' files installed", 
changing the language to "JSON" in vs code also does not help because formatting payload breaks header(s) (I have variable "Authorization: Bearer {{$global.token}}"), 
I am making a lot JSON and XML request payload modifications and now I am fixing formatting manually  - it is a bit uncomfortable.
Do you have any suggestions on how to deal with this inconvenience?

thanks,
Tomasz

**Comment by AnWeber** [comment]
I don't have a solution myself yet. Exactly the variables force me to build my own formatter and I don't have the time for that right now
